Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -382,7 +389,15 @@ public struct PackageBuilder {
             throw ModuleError.modulesNotFound(missingModuleNames)
         }
 
-        return modules
+        // Normally, test modules are only dependent upon modules with
+        // the same basename. For example, a test module in
+        // 'Root/Tests/FooTests' is dependent upon 'Root/Sources/Foo'.
+        // Only do this if there is no explict dependency declared in manifest.
```

Comment:
Can we make `module.dependencies` a `[String]?` so instead of empty we use "specified at all"? That way explicit specification is always complete and honored with no other magic.